independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ActivateRequestContextInterceptor.java

```diff
@@ -38,14 +39,16 @@ private CompletionStage<?> invokeStage(InvocationContext ctx) {
         }
 
         return activate(requestContext)
-                .thenCompose(v -> proceedWithStage(ctx))
-                .whenComplete((r, t) -> requestContext.terminate());
+                .thenCompose(state -> proceedWithStage(ctx).whenComplete((r, t) -> {
+                    requestContext.destroy(state);
+                    requestContext.deactivate();
+                }));
```

Comment:
I was wondering what's the difference here, as `terminate()` == `destroy()` + `deactivate()`. It's that you explicitly pass `state` to `destroy`, right?

Reply:
Ah all changes in this file do this same thing. I wonder if it's just a precaution, or if `destroy()` without parameters finds (and destroys) a different context than it should...

In any case, being explicit can't hurt.